polar-core/src/kb.rs

```diff
@@ -239,15 +239,17 @@ impl KnowledgeBase {
                     Ok(RuleParamMatch::True)
                 }
             } else {
-                Err(error::OperationalError::InvalidState{msg: format!(
-                    "All registered classes must have a registered MRO. Class {} does not have a registered MRO.",
-                    &rule_instance.tag
-                )}.into())
+                // If `rule_instance.tag` were registered as a class, it would have an MRO.
+                Ok(RuleParamMatch::False(format!(
+                    "Rule specializer {} on parameter {} is not registered as a class.",
+                    rule_instance.tag, index
+                )))
             }
         } else {
-            // TODO(gj): `rule_type_instance.tag` was registered as something other than an
-            // external instance. What should we do here?
-            unreachable!("Unregistered specializer classes should be caught before this point.");
+            Ok(RuleParamMatch::False(format!(
```

Comment:
Same question here.

Reply:
Could this message say `instead registered as a constant with value: {}`? That's the only other possible case, right?

Reply:
5b365d63